src/App.js

```diff
@@ -80,11 +77,12 @@ class App extends Component {
         />
 
         <NewUser
-          showModal={() => this.showUserModal()}
+          showModal={!this.state.username}
           onSubmit={this.saveUser}
         />
         <div className="App-sidebar">
           <h2>Chat App</h2>
+          {username && <p>{username} is logged in</p>}
```

Comment:
if a username is non-existent, then the p tag will not be rendered. The `&&` is shorthand and this can all be written like so:
```
{username ? <p>{username} is logged in</p> : null}
```